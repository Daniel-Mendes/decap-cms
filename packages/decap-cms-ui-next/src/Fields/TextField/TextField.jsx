import React, { useState, forwardRef } from 'react';
import styled from '@emotion/styled';

import Field from '../../Field';

export const StyledInput = styled.input`
  color: ${({ theme }) => theme.color.highEmphasis};
  background: none;
  border: none;
  outline: none;
  width: calc(100% + 32px);
  font-family: inherit;
  font-size: ${({ title }) => (title ? '2rem' : '1rem')};
  font-weight: ${({ title }) => (title ? 'bold' : 'normal')};
  letter-spacing: ${({ title }) => (title ? '-0.5px' : '0')};
  line-height: 1rem;
  caret-color: ${({ theme, error }) =>
    error ? theme.color.danger['900'] : theme.color.primary['800']};
  margin: -2rem -1rem -1rem -1rem;
  padding: 2rem 1rem 1rem 1rem;
  ${({ clickable }) => (clickable ? `cursor: pointer;` : ``)} ::placeholder {
    color: ${({ theme }) => theme.color.disabled};
  }
`;

const TextField = forwardRef(
  (
    {
      name,
      label,
      icon,
      description,
      status,
      onChange,
      onClick,
      readOnly,
      placeholder,
      children,
      value,
      focus,
      title,
      type,
      password,
      className,
      inline,
      error,
      errors,
    },
    ref,
  ) => {
    const [inputFocus, setInputFocus] = useState();

    return (
      <Field
        onClick={onClick}
        label={label}
        labelTarget={name}
        description={description}
        status={status}
        focus={focus || inputFocus}
        className={className}
        inline={inline}
        error={error}
        errors={errors}
        icon={icon}
        clickable={!readOnly}
      >
        <StyledInput
          ref={ref}
          clickable={readOnly && !!onClick}
          onClick={onClick}
          readOnly={readOnly}
          type={password ? 'password' : type ? type : 'text'}
          id={name}
          name={name}
          placeholder={placeholder ? placeholder : label ? `Type ${label.toLowerCase()} here` : ''}
          value={value || ''}
          onChange={e => onChange(e)}
          onFocus={() => setInputFocus(true)}
          onBlur={() => setInputFocus(false)}
          autoComplete="off"
          title={title}
          error={error}
        />
        {children && children}
      </Field>
    );
  },
);

TextField.displayName = 'TextField';

export default TextField;
