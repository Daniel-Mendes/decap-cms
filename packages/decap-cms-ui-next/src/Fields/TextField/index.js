export { default, StyledInput } from './TextField';
