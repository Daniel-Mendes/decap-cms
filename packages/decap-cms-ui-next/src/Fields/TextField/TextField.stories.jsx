import React from 'react';
import styled from '@emotion/styled';
import { useArgs } from '@storybook/preview-api';

import TextField from '.';

const StyledTextField = styled(TextField)`
  width: 33vw;
`;

export default {
  title: 'Fields/TextField',
  component: TextField,
  argTypes: {
    errors: {
      control: 'object',
      if: {
        arg: 'error',
      },
    },
  },
  args: {
    label: 'Text Field Label',
    placeholder: 'Type something here',
    description: 'This is a description',
    status: 'Required',
    value: '',
    title: false,
    filled: false,
    error: false,
    errors: [
      {
        message: 'Error message.',
      },
    ],
  },
};

export function _TextField(args) {
  const [{ value }, updateArgs] = useArgs();

  function handleChange(e) {
    updateArgs({ value: e.target.value });
  }

  return <StyledTextField {...args} value={value} onChange={handleChange} />;
}
