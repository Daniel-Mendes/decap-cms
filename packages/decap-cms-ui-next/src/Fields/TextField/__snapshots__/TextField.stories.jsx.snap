// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields/TextField _TextField smoke-test 1`] = `"<div class="e11i8s480 css-gzruzu-FieldWrap-StyledTextField e1x0ipyd6"><div class="css-1uqy5pq-FieldInside-FieldInside-FieldInside-FieldInside e1x0ipyd5"><label class="css-1n9k11u-Label-StyledLabel e1x0ipyd3">Text Field Label <div color="neutral" radius="large" class="css-6930ng-BadgeWrap-BadgeWrap epfx4ki3">Required</div></label><div class="css-dj548v-ChildrenWrap e1x0ipyd4"><input type="text" placeholder="Type something here" autocomplete="off" class="css-h0pblb-StyledInput e4jt8f0" value=""></div><p class="css-nalsj0-StyledDescription e1x0ipyd1">This is a description</p></div></div>"`;
